src/megaclient.cpp

```diff
@@ -3782,26 +3835,42 @@ bool MegaClient::isFetchingNodesPendingCS()
 #ifdef ENABLE_SYNC
 void MegaClient::resumeResumableSyncs()
 {
-
-    if (!syncConfigs || !allowAutoResumeSyncs)
+    if (!syncConfigs)
     {
         return;
     }
     for (const auto& config : syncConfigs->all())
     {
-        if (!config.isResumable())
-        {
-            continue;
-        }
-        const auto e = addsync(config, DEBRISFOLDER, nullptr);
-        if (e == 0)
+        SyncError syncError = static_cast<SyncError>(config.getError());
+        syncstate_t newstate = isAnError(syncError) ? SYNC_FAILED : SYNC_DISABLED;
+        if (config.isResumableAtStartup())
         {
-            app->sync_auto_resumed(config.getLocalPath(), config.getRemoteNode(),
-                                   static_cast<long long>(config.getLocalFingerprint()),
-                                   config.getRegExps());
+            if (!mFirstSyncResumed)
```

Comment:
Do you think it could be replaced by a local variable to this method, rather than another member in MegaClient?

Reply:
Yes, thus we don't need to reset it elsewhere. good point